python/tank/templatekey.py

```diff
@@ -256,6 +273,145 @@ def _as_string(self, value):
         return value if isinstance(value, basestring) else str(value)
 
 
+class TimestampKey(TemplateKey):
+    """
+    Key whose value is a time string formatted with strftime.
+    """
+
+    def __init__(
+        self,
+        name,
+        utc_default=False,
+        format_spec="%Y-%m-%d-%H-%M-%S"
+    ):
+        """
+        Constructor
+        :param name: Name by which the key will be refered.
+        :param utc_default: If True, the default value will be an UTC time. Defaults to False.
+        :param format_spec: Specification for formating when casting to/from a string.
+                            The format follows the convention of strftime and strptime. The
+                            default value is "%Y-%m-%d-%H-%M-%S". Given June 24th, 2015 at
+                            9:20:30 PM, this will yield 2015-06-24-21-20-30
+        """
+        if isinstance(format_spec, basestring) is False:
+            raise TankError("format_spec for TimestampKey %s is not of type string: %s" %
```

Comment:
Chicken an egg problem. Invoking __repr__ requires that the base class is initialized (needs a self.name to be set), but I need to initialize the format_spec and the utc_default members so that validation of the default value works in the base class.